src/mongo/client/dbclient_rs.cpp

```diff
@@ -417,10 +418,23 @@ void DBClientReplicaSet::_auth(const BSONObj& params) {
                 break;
             }
 
-            conn->auth(params);
+            BSONObj interposedParams;
```

Comment:
Why not just start with

BSONObj interposedParams = params;

? Then you woudn't need the else block below on line 430